src/dataaggregator/aggregator/policyPoller_test.go

```diff
@@ -0,0 +1,96 @@
+package aggregator_test
+
+import (
+	"code.cloudfoundry.org/lager"
+	. "dataaggregator/aggregator"
+	"dataaggregator/aggregator/fakes"
+	. "dataaggregator/policy"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"time"
+)
+
+var _ = Describe("PolicyPoller", func() {
+	var (
+		database *fakes.FakeDB
+
+		poller     *PolicyPoller
+		testAppId1 = "testAppId"
+		policyStr1 = `
+		{
+		   "instance_min_count":1,
+		   "instance_max_count":5,
+		   "scaling_rules":[
+		      {
+		         "metric_type":"MemoryUsage",
+		         "stat_window_secs":300,
+		         "breach_duration_secs":300,
+		         "threshold":30,
+		         "operator":"<",
+		         "cool_down_secs":300,
+		         "adjustment":"-1"
+		      }
+		   ]
+		}`
+	)
+
+	BeforeEach(func() {
+		database = &fakes.FakeDB{}
+		logger := lager.NewLogger("PolicyPoller-test")
+		poller = NewPolicyPoller(logger, TestPolicyPollerInterval, database)
+	})
+	Context("Start", func() {
+		BeforeEach(func() {
+			poller.Start()
+		})
+		AfterEach(func() {
+			poller.Stop()
+		})
+		Context("when the poller is started", func() {
+			BeforeEach(func() {
+				database.RetrievePoliciesStub = func() ([]*PolicyJson, error) {
+					return []*PolicyJson{&PolicyJson{AppId: testAppId1, PolicyStr: policyStr1}}, nil
+				}
+			})
+			It("should retrieve policies for every interval", func() {
+				Eventually(database.RetrievePoliciesCallCount, 2500*time.Millisecond, 500*time.Millisecond).Should(Equal(2))
+			})
+		})
+		Context("when there is no policy", func() {
+			BeforeEach(func() {
+				database.RetrievePoliciesStub = func() ([]*PolicyJson, error) {
+					return []*PolicyJson{}, nil
+				}
+			})
+			It("no trigger generated", func() {
+				Consistently(func() int { return len(poller.PolicyMap) }, 2500*time.Millisecond, 500*time.Millisecond).Should(Equal(0))
```

Comment:
For this story "Policies are periodically retrieved by data aggregator", we just retrieve policies and parse triggers. We will go to next step to aggregate the memory datas.

Now when I use ginkgo -r -race , it will show some warning msg as below. Seems Ben has the same issue. Could you give some advice?
Warning msg:
WARNING: DATA RACE
Read by goroutine 10:
  dataaggregator/aggregator/fakes.(*FakeDB).RetrievePolicies()
      /Users/qiyang/dev/openjavaspace/app-autoscaler/src/dataaggregator/aggregator/fakes/fake_db.go:33 +0x22d
  dataaggregator/aggregator.(*PolicyPoller).retrievePolicies()
      /Users/qiyang/dev/openjavaspace/app-autoscaler/src/dataaggregator/aggregator/policyPoller.go:59 +0x7c
  dataaggregator/aggregator.(*PolicyPoller).startPolicyRetrieve()
      /Users/qiyang/dev/openjavaspace/app-autoscaler/src/dataaggregator/aggregator/pol